drivers/adc/ad4170/ad4170.h

```diff
@@ -0,0 +1,1142 @@
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef AD4170_H_
+#define AD4170_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "no_os_util.h"
+#include "no_os_gpio.h"
+#include "no_os_spi.h"
+
+#ifndef ECOMM
+#define ECOMM 70
+#endif
+
+#define AD4170_R1B				(1ul << 16)
+#define AD4170_R2B				(2ul << 16)
+#define AD4170_R3B				(3ul << 16)
+#define AD4170_R4B				(4ul << 16)
+#define AD4170_TRANSF_LEN(x)			((x) >> 16)
+#define AD4170_ADDR(x)				((x) & 0xFFFF)
+
+#define AD4170_SPI_SYNC_PATTERN			0x2645
+
+#define AD4170_REG_READ_6(x)			(((x) & 0x3F) | 0x40)
+#define AD4170_REG_WRITE_6(x)			((x) & 0x3F)
+#define AD4170_REG_READ_14(x)			(((x) & 0x3FFF) | 0x4000)
+#define AD4170_REG_WRITE_14(x)			((x) & 0x3FFF)
+#define AD4170_REG_INTERFACE_CONFIG_A		(AD4170_R1B | 0x00)
+#define AD4170_REG_INTERFACE_CONFIG_B		(AD4170_R1B | 0x01)
+#define AD4170_REG_DEVICE_CONFIG		(AD4170_R1B | 0x02)
+#define AD4170_REG_CHIP_TYPE			(AD4170_R1B | 0x03)
+#define AD4170_REG_PRODUCT_ID_L			(AD4170_R1B | 0x04)
+#define AD4170_REG_PRODUCT_ID_H			(AD4170_R1B | 0x05)
+#define AD4170_REG_CHIP_GRADE			(AD4170_R1B | 0x06)
+#define AD4170_REG_SCRATCH_PAD			(AD4170_R1B | 0x0a)
+#define AD4170_REG_SPI_REVISION			(AD4170_R1B | 0x0b)
+#define AD4170_REG_VENDOR_L			(AD4170_R1B | 0x0c)
+#define AD4170_REG_VENDOR_H			(AD4170_R1B | 0x0d)
+#define AD4170_REG_INTERFACE_CONFIG_C		(AD4170_R1B | 0x10)
+#define AD4170_REG_INTERFACE_STATUS_A		(AD4170_R1B | 0x11)
+#define AD4170_REG_DATA_STATUS			(AD4170_R2B | 0x14)
+#define AD4170_REG_DATA_16b			(AD4170_R2B | 0x16)
+#define AD4170_REG_DATA_16b_STATUS		(AD4170_R3B | 0x18)
+#define AD4170_REG_DATA_24b			(AD4170_R3B | 0x1c)
+#define AD4170_REG_DATA_24b_STATUS		(AD4170_R4B | 0x20)
+#define AD4170_REG_DATA_32b			(AD4170_R4B | 0x24)
+#define AD4170_REG_DATA_PER_CHANNEL(ch)		(AD4170_R4B | (0x28 + 4 * (ch)))
+#define AD4170_REG_PIN_MUXING			(AD4170_R2B | 0x68)
+#define AD4170_REG_CLOCK_CTRL			(AD4170_R2B | 0x6a)
+#define AD4170_REG_STANDBY_CTRL			(AD4170_R2B | 0x6c)
+#define AD4170_REG_POWER_DOWN_SW		(AD4170_R2B | 0x6e)
+#define AD4170_REG_ADC_CTRL			(AD4170_R2B | 0x70)
+#define AD4170_REG_ERROR_EN			(AD4170_R2B | 0x72)
+#define AD4170_REG_ERROR			(AD4170_R2B | 0x74)
+#define AD4170_REG_INFO1			(AD4170_R2B | 0x76) /* TODO: implement this when it's specified in doc. */
```

Comment:
Should we remove the TODO before release?

Reply:
good point, turns out INFO1 is not in the latest datasheet, and is also unused in the driver so I'm removing it